### PR TITLE
feat: (home) Make home banner automatically appear when ifo or auction is alive or soon

### PR DESCRIPTION
**Pull request by memoyil**
When ifo or farm auction is live or soon home banner will appear automatically

### CODE REVIEW

File: src/views/Home/components/Banners/IFOBanner.tsx [hunk 1]
```diff
@@ -105,7 +106,15 @@ const IFOBanner = () => {
           </NextLinkFromReactRouter>
         </LeftWrapper>
         <RightWrapper>
-          <img src="/images/decorations/3d-ifo-era.png" alt="IFO Era" />
+          <img
+            src={`/images/decorations/3d-ifo-${id}.png`}
```
Comment (memoyil) [inline src/views/Home/components/Banners/IFOBanner.tsx:110]: image should be in this format

File: src/views/Home/components/Banners/FarmAuctionsBanner.tsx [hunk 2]
```diff
@@ -82,12 +89,39 @@ const RightWrapper = styled.div`
...
 const FarmAuctionsBanner = () => {
   const { t } = useTranslation()
+  const currentBlock = useCurrentBlock()
 
-  return (
+  const farmAuctionContract = useFarmAuctionContract(false)
+
+  const { data: currentAuction = null } = useSWRImmutable(['farmAuction', 'currentAuction'], async () => {
+    const auctionId = await farmAuctionContract.currentAuctionId()
+    const [auctionData, contenders] = await multicallv2(
```
Comment (chefilip) [inline src/views/Home/components/Banners/FarmAuctionsBanner.tsx:98]: could we use `useSWRMulticall`  in this case ?

Ideally, `multicallv2` shouldn't be used in the Component-level.

File: src/views/Home/components/Banners/IFOBanner.tsx [hunk 3]
```diff
@@ -86,15 +94,43 @@ const RightWrapper = styled.div`
...
+  const isIfoAlive = !!(activeIfo && currentBlock && currentIfoBlocks && currentIfoBlocks.endBlock > currentBlock)
 
-  return (
+  return isIfoAlive ? (
     <Wrapper>
       <Inner>
         <LeftWrapper>
-          <StyledSubheading>{t('Live')}</StyledSubheading>
-          <StyledHeading scale="xl">ERA IFO</StyledHeading>
+          <StyledSubheading>{currentIfoBlocks.startBlock < currentBlock ? t('Live') : t('Soon')}</StyledSubheading>
```
Comment (chefilip) [inline src/views/Home/components/Banners/IFOBanner.tsx:132]: Screen Shot 2022-02-21 at 16 34 22

Could we reuse this function to get status ?

File: src/views/Home/components/Banners/IFOBanner.tsx [hunk 4]
```diff
@@ -86,15 +94,43 @@ const RightWrapper = styled.div`
   }
 `
 
+const activeIfo = ifosConfig.find((ifo) => ifo.isActive)
+
 const IFOBanner = () => {
   const { t } = useTranslation()
+  const currentBlock = useCurrentBlock()
+
+  const { data: currentIfoBlocks = { startBlock: 0, endBlock: 0 } } = useSWRImmutable(
```
Comment (chefilip) [inline src/views/Home/components/Banners/IFOBanner.tsx:103]: Could we make this hook reusable ? I'm going to need it for updating the status in the IFO menu item

File: src/views/Home/components/Banners/IFOBanner.tsx (hunk 3 above)
Comment (memoyil) [reply src/views/Home/components/Banners/IFOBanner.tsx:132]: Done

File: src/views/Home/components/Banners/IFOBanner.tsx (hunk 4 above)
Comment (memoyil) [reply src/views/Home/components/Banners/IFOBanner.tsx]: Done

File: src/views/Home/components/Banners/FarmAuctionsBanner.tsx (hunk 2 above)
Comment (memoyil) [reply src/views/Home/components/Banners/FarmAuctionsBanner.tsx]: Done

File: src/views/Home/components/Banners/IFOBanner.tsx [hunk 5]
```diff
@@ -1,7 +1,16 @@
+import { memo } from 'react'
 import styled from 'styled-components'
 import { Text, Flex, Button, ArrowForwardIcon, Heading } from '@pancakeswap/uikit'
 import { NextLinkFromReactRouter } from 'components/NextLink'
 import { useTranslation } from 'contexts/Localization'
+import useSWRImmutable from 'swr/immutable'
+import ifoV3Abi from 'config/abi/ifoV3.json'
```
Comment (chefilip) [inline src/views/Home/components/Banners/IFOBanner.tsx:7]: remove if no used. `Eslint` should warn this

Comment (memoyil) [reply src/views/Home/components/Banners/IFOBanner.tsx]: Done

**Review by chefilip — APPROVED** [review]

**Pull request merged**